multisig/simulations/ethereum/vipConverter/index.ts

```diff
@@ -0,0 +1,306 @@
+      const usdtBalanceUsdtPrimeCurrent = await usdt.balanceOf(PLP);
+
+      expect(usdtBalanceUsdtPrimeCurrent).to.equal(usdtBalanceUsdtPrimePrevious.add(amount));
+      expect(usdcBalanceUsdtPrimeConverterCurrent).to.closeTo(
+        usdcBalanceUsdtPrimeConverterPrevious.sub(amount),
+        parseUnits("1", 6),
+      );
+      expect(await usdc.balanceOf(destinationAddressForUsdcConverter)).to.closeTo(
+        plpBalanceForUsdcPrevious.add(amount),
+        parseUnits("1", 6),
+      );
+    });
+  });
+});
+
+forking(20120060, () => {
+  const provider = ethers.provider;
+  let prime: Contract;
+  let plp: Contract;
+
+  describe("Pre-VIP behaviour", () => {
+    before(async () => {
+      prime = new ethers.Contract(PRIME, PRIME_ABI, provider);
+      plp = new ethers.Contract(PLP, PLP_ABI, provider);
+    });
+
+    it("Plp should not contain tokens", async () => {
+      plp = new ethers.Contract(PLP, PLP_ABI, provider);
+
+      expect(await plp.lastAccruedBlockOrSecond(USDT)).to.be.equal(0);
+      expect(await plp.lastAccruedBlockOrSecond(USDC)).to.be.equal(0);
+      expect(await plp.lastAccruedBlockOrSecond(WETH)).to.be.equal(0);
+      expect(await plp.lastAccruedBlockOrSecond(WBTC)).to.be.equal(0);
+    });
+
+    it("Prime should have not contain number of revocable and irrevocable tokens", async () => {
+      expect(await prime.irrevocableLimit()).to.equal(0);
+      expect(await prime.revocableLimit()).to.equal(0);
+    });
+  });
+
+  describe("Post-VIP behavior", () => {
+    let prime: Contract;
+    let plp: Contract;
+
+    before(async () => {
+      await pretendExecutingVip(vipConverter());
+
+      prime = new ethers.Contract(PRIME, PRIME_ABI, provider);
+      plp = new ethers.Contract(PLP, PLP_ABI, provider);
+    });
+
+    it("Comptroller lst should have correct Prime token address", async () => {
+      const comptrollerLst = new ethers.Contract(COMPTROLLER_LST, COMPTROLLER_ABI, provider);
+      expect(await comptrollerLst.prime()).to.be.equal(PRIME);
+    });
+
+    it("Prime should contain correct markets", async () => {
+      expect((await prime.markets(VUSDC_CORE))[4]).to.be.equal(true);
+      expect((await prime.markets(VUSDT_CORE))[4]).to.be.equal(true);
+      expect((await prime.markets(VWBTC_CORE))[4]).to.be.equal(true);
+      expect((await prime.markets(VWETH_LST))[4]).to.be.equal(true);
+    });
+
+    it("Prime should have correct number of revocable and irrevocable tokens", async () => {
+      expect(await prime.irrevocableLimit()).to.equal(0);
+      expect(await prime.revocableLimit()).to.equal(500);
+    });
+
+    it("Plp should have correct tokens", async () => {
+      expect(await plp.lastAccruedBlockOrSecond(USDT)).to.be.gt(0);
+      expect(await plp.lastAccruedBlockOrSecond(USDC)).to.be.gt(0);
+      expect(await plp.lastAccruedBlockOrSecond(WETH)).to.be.gt(0);
+      expect(await plp.lastAccruedBlockOrSecond(WBTC)).to.be.gt(0);
+    });
+  });
+});
```

Comment:
I will suggest to add some storage layout checks for new implementation of single token converter

Reply:
There is not any change in the storage layout, we have only changed the `approve` by `forceApprove`.
Also the tests written regarding conversion and others execute on the new implementation storage. Wdsy?